packages/core/src/provisioning.ts

```diff
@@ -116,7 +116,7 @@ export class AccountProvisioning extends Storable {
 
     accountId?: AccountID = undefined;
 
-    status: ProvisioningStatus = ProvisioningStatus.Unprovisioned;
```

Comment:
@MaKleSoft I've reverted this because otherwise the default `simple` provisioning backend won't work, and overriding the status for tests seemed brittle. I'll instead add a new server setting to allow/disallow signup, which should give more flexibility to people who need automatic provisioning and also allow non-provisioned accounts to be created (assuming the server is behind a VPN, for example).

Reply:
Makes sense. I would add maybe add a setting like `PTC_PROVISIONING_BASIC_DEFAULT_STATUS` for the `BasicProvisioner`, but for `DirectoryProvisioner` it is better to "hardcode" the default status by overriding [`_getOrCreateAccountProvisioning`](https://github.com/padloc/padloc/blob/v4/packages/core/src/provisioning.ts#L332)

Reply:
Or, even better, add a protected method `_getDefaultAccountProvisioning` to `BasicProvisioner`, which would be consumed [here](https://github.com/padloc/padloc/blob/feature/scim-v1/packages/core/src/provisioning.ts#L328) and  could then be overridden by `DirectoryProvisioner`.

Reply:
I've added the new setting to disable signup, and tests should be passing again. Do you still want me to add this new method for the default?